pkg/core/statesync/neotest_test.go

```diff
@@ -419,8 +467,12 @@ func TestStateSyncModule_RestoreBasicChain(t *testing.T) {
 			}
 			require.Error(t, module.AddBlock(b))
 		})
+		start := uint32(stateSyncPoint - maxTraceable + 1)
+		if enableStorageSync {
+			start = 1
+		}
 
-		for i := uint32(stateSyncPoint - maxTraceable + 1); i <= stateSyncPoint; i++ {
+		for i := start; i <= stateSyncPoint; i++ {
 			b, err := bcSpout.GetBlock(bcSpout.GetHeaderHash(i))
 			require.NoError(t, err)
 			require.NoError(t, module.AddBlock(b))
```

Comment:
Prior to blocks addition you need to check that module's block height is properly initialized and matches expectations. Also, prior to storage sync you need to check that `BlockHeight()` API panics.